src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -473,18 +466,17 @@ private static bool TryParseValue(ref Utf8JsonReader reader, out JsonDocument do
                             int payloadLength = reader.ValueSpan.Length + 2;
                             Debug.Assert(payloadLength > 1);
 
-                            int openQuote = checked((int)startingOffset) - payloadLength;
                             ReadOnlySpan<byte> readerSpan = reader.OriginalSpan;
 
                             Debug.Assert(
-                                readerSpan[openQuote] == (byte)'"',
-                                $"Calculated span starts with {readerSpan[openQuote]}");
+                                readerSpan[(int)reader.TokenStartIndex] == (byte)'"',
+                                $"Calculated span starts with {readerSpan[(int)reader.TokenStartIndex]}");
 
                             Debug.Assert(
-                                readerSpan[(int)startingOffset - 1] == (byte)'"',
-                                $"Calculated span ends with {readerSpan[(int)startingOffset - 1]}");
+                                readerSpan[(int)reader.TokenStartIndex + payloadLength - 1] == (byte)'"',
+                                $"Calculated span ends with {readerSpan[(int)reader.TokenStartIndex + payloadLength - 1]}");
 
-                            valueSpan = readerSpan.Slice(openQuote, payloadLength);
+                                valueSpan = readerSpan.Slice((int)reader.TokenStartIndex, payloadLength);
```

Comment:
Nit: Seems like an extra (out-of-alignment) indentation got added here.

Reply:
The case structure with {} has weird alignment. I would prefer if we just align it to what VS recommend and then such issues won't creep up.

When I was moving the Debug.Asserts around, I had to keep fighting the alignment. Are you ok with that?

Effectively, it will be indented by one:

```C#
case foo:
   {

   }
case bar:
   {
   }
```

Alternatively, we remove the braces all together and use the case/break for scope (and then change local variable names).

Reply:
I'd prefer to just restore the alignment on this line, since it keeps the change limited to what actually changed.  (Clearly something on my profile sets the braces as aligned with the case keyword, which is honestly what I prefer... if that's an odd-man-out style then normalizing it should (ideally) be separate from functional changes (or part of a big rewrite)).

Reply:
> which is honestly what I prefer... if that's an odd-man-out style then normalizing it should (ideally) be separate from functional changes (or part of a big rewrite)).

I don't have a preference either way except that I prefer the defaults. It can be a separate change.